db/write_batch.cc

```diff
@@ -1854,7 +1864,10 @@ Status WriteBatch::VerifyChecksum() const {
     if (checksum_protected) {
       s = prot_info_->entries_[prot_info_idx++]
               .StripC(column_family)
-              .StripKVO(key, value, static_cast<ValueType>(tag))
+              .StripKVO(SliceParts(&key, 1),
```

Comment:
I'm concerned about performance overhead of this extra complexity even when TimedPut is not used.

Perhaps instead if we pass nullptr for `unix_write_time` to `ReadRecordFromWriteBatch` it would include the write time as part of the value, for simpler handling in cases like protection info.

Reply:
Thanks for this idea!